Tests/Marketplace/search_and_install_packs.py

```diff
@@ -86,143 +86,88 @@ def is_pack_hidden(pack_id: str) -> bool:
     """
     metadata_path = os.path.join(PACKS_FULL_PATH, pack_id, PACK_METADATA_FILENAME)
     if pack_id and os.path.isfile(metadata_path):
-        with open(metadata_path, 'r') as json_file:
+        with open(metadata_path) as json_file:
             pack_metadata = json.load(json_file)
             return pack_metadata.get('hidden', False)
     else:
         logging.warning(f'Could not open metadata file of pack {pack_id}')
     return False
 
 
-def create_dependencies_data_structure(response_data: dict, dependants_ids: list, dependencies_data: list,
-                                       checked_packs: list):
-    """ Recursively creates the packs' dependencies data structure for the installation requests
-    (only required and uninstalled).
+def create_dependencies_data_structure(response_data: dict, dependants_ids: list, dependencies_data: list, checked_packs: list):
+    """
+    Recursively create packs' dependencies data structure for installation requests (only required and uninstalled).
 
     Args:
-        response_data (dict): The GET /search/dependencies response data.
+        response_data (dict): Dependencies data from the '/search/dependencies' endpoint response.
         dependants_ids (list): A list of the dependant packs IDs.
         dependencies_data (list): The dependencies data structure to be created.
         checked_packs (list): Required dependants that were already found.
     """
-
     next_call_dependants_ids = []
 
     for dependency in response_data:
         dependants = dependency.get('dependants', {})
-        for dependant in dependants.keys():
+        for dependant in dependants:
             is_required = dependants[dependant].get('level', '') == 'required'
-            if dependant in dependants_ids and is_required and dependency.get('id') not in checked_packs:
-                dependencies_data.append({
-                    'id': dependency.get('id'),
-                    'version': dependency.get('extras', {}).get('pack', {}).get('currentVersion')
-                })
-                next_call_dependants_ids.append(dependency.get('id'))
-                checked_packs.append(dependency.get('id'))
+            if dependant in dependants_ids and is_required and dependency['id'] not in checked_packs:
+                dependencies_data.append(dependency)
```

Comment:
How does the "dependency" object look now? With all data? not only ID and version?

Is the 'version': dependency.get('extras', {}).get('pack', {}).get('currentVersion') line handled?